### PR TITLE
updated regionprops_table test for iterable properties

### PR DESCRIPTION
**Pull request by yfukai**
## Description

Associated with pull request #5239 and issue #5222 . I updated test `test_regionprops_table_equal_to_original` so that they compare not only non-iterable properties but also iterable ones of `regionprops` and `regionprops_table`.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
→ not applicable
- Gallery example in `./doc/examples` (new features only)
→ not applicable
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
→ not applicable
- Unit tests
→ one modified here
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
→ checked with autopep8

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Review by mkcor — COMMENTED** [review]
Excellent, thank you! With the breakdown introduced by `separator="-"`, I'm not seeing any instance where `i > 0`... Please correct me if I'm missing something!
It seems to me that all properties are covered by this main for loop:
```py
    for prop, dtype in COL_DTYPES.items():
        for reg in regions:
            rp = reg[prop]
            if np.isscalar(rp) or \
                    prop in OBJECT_COLUMNS or \
                    dtype is np.object_:
                assert_array_equal(rp, out_table[prop][0])
            else:
                shape = rp.shape if isinstance(rp, np.ndarray) else (len(rp),)
                for ind in np.ndindex(shape):
                    modified_prop = "-".join(map(str, (prop,) + ind))
                    loc = ind if len(ind) > 1 else ind[0]
                    assert_equal(rp[loc], out_table[modified_prop][0])
```

**Comment by yfukai** [comment]
Hi, thank you for your comment @mkcor ! Yeah you're right, `SAMPLE` has only one region with the label `1`, so `i>0` doesn't appear in the loop for now. However, I have to note that I think the modification you suggested would result in an error when someone add label `2` to `SAMPLE`. Instead we may be able to fix the index to `0` as follows to make the code a bit simpler:
```python
reg=regions[0]
for prop, dtype in COL_DTYPES.items():
    rp = reg[prop]
    if np.isscalar(rp) or \
            prop in OBJECT_COLUMNS or \
            dtype is np.object_:
        assert_array_equal(rp, out_table[prop][0])
    else:
        shape = rp.shape if isinstance(rp, np.ndarray) else (len(rp),)
        for ind in np.ndindex(shape):
            modified_prop = "-".join(map(str, (prop,) + ind)) 
            loc = ind if len(ind) > 1 else ind[0]
            assert_equal(rp[loc], out_table[modified_prop][0])
```

**Comment by mkcor** [comment]
@yfukai ok, very well. This way your test is robust to the case where `SAMPLE` would have more than one region. Fair enough! Then I'm happy to approve the code as is.

**Review by mkcor — APPROVED** [review]

**Comment by yfukai** [comment]
Wonderful, thanks!

File: skimage/measure/tests/test_regionprops.py [hunk 1]
```diff
@@ -580,12 +579,22 @@ def test_regionprops_table():
...
     regions = regionprops(SAMPLE, INTENSITY_FLOAT_SAMPLE)
     out_table = regionprops_table(SAMPLE, INTENSITY_FLOAT_SAMPLE,
-                                  properties=COL_DTYPES.keys())
-
-    for prop in COL_DTYPES.keys():
-        for i in range(len(regions)):
-            if not isinstance(regions[i][prop], Iterable):
-                assert regions[i][prop] == out_table[prop][i]
+                                  properties=COL_DTYPES.keys(),
+                                  separator="-")
```
Comment (rfezzani) [inline skimage/measure/tests/test_regionprops.py:583]:
```suggestion
                                  properties=COL_DTYPES.keys())
```

isn't `"-"` the default value for the `separator` argument?

File: skimage/measure/tests/test_regionprops.py [hunk 2]
```diff
@@ -580,12 +579,22 @@ def test_regionprops_table():
...
+
+    for prop, dtype in COL_DTYPES.items():
+        for i, reg in enumerate(regions):
+            rp = reg[prop]
+            if np.isscalar(rp) or \
+                    prop in OBJECT_COLUMNS or \
+                    dtype is np.object_:
```
Comment (rfezzani) [inline skimage/measure/tests/test_regionprops.py:590]:
```suggestion
            if np.isscalar(rp) or prop in OBJECT_COLUMNS:
```
The `dtype is np.object_` condition is not necessary.

**Review by rfezzani — COMMENTED** [review]
Thank you @yfukai! The PR is really good. I just left two small suggestions :wink:

File: skimage/measure/tests/test_regionprops.py (hunk 2 above)
Comment (yfukai) [reply skimage/measure/tests/test_regionprops.py:589]: Oh thanks, I see! This is actually (dirty) copy from the _regionprops.py logic itself. I'll try to rewrite it in a bit more DRY way.

File: skimage/measure/tests/test_regionprops.py (hunk 1 above)
Comment (yfukai) [reply skimage/measure/tests/test_regionprops.py]: Exactly I noticed it is, thank you for pointing this out!

**Review by jni — APPROVED** [review]

**Comment by jni** [comment]
Awesome, thank you again @yfukai!

**Comment by yfukai** [comment]
Thanks for merging @jni , but I think I still haven't dealt with @rfezzani 's second comment. Maybe I will make some new PR for that at some point!

**Comment by jni** [comment]
Oops! :sweat_smile: I saw the comment and the commit and made an assumption. :grimacing: Future PRs are absolutely welcome! I think this is already a major improvement anyway. =)

**Comment by rfezzani** [comment]
:smile: no worry, it is not a big deal! The PR is already in really good shape, but as @jni said, future PRs are welcome! Thank you again @yfukai :wink:

**Pull request merged**